### PR TITLE
Added AWS HA RDS deploy guidance

### PR DESCRIPTION
**Pull request by cwarnermm**
Added AWS RDS HA config guidance to the following product docs pages:
- [HA Cluster > Database Configuration](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6967/scale/high-availability-cluster.html#database-configuration)
- [Data Source](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6967/configure/environment-configuration-settings.html#database-datasource), [Data Source Replicas](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6967/configure/environment-configuration-settings.html#database-readreplicas), and [Data Source Search Replicas](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6967/configure/environment-configuration-settings.html#database-searchreplicas) configuration setting docs

Are there additional docs sections that would benefit from links into this new guidance?

### CODE REVIEW

File: source/configure/database-configuration-settings.rst [hunk 1]
```diff
@@ -139,7 +139,14 @@ Data source
...
 | +----------------------------------------+-----------------+---------------------------------------------------------------------------+ |
 |                                                                                                                                          |
-+---------------------------------------------------------------+--------------------------------------------------------------------------+
++------------------------------------------------------------+-----------------------------------------------------------------------------+
+| **AWS High Availablity RDS cluster deployments**                                                                                         |
+|                                                                                                                                          |
+| For an AWS High Availability RDS cluster deployment, point this configuration setting to the write/read endpoint at the **cluster**      |
+| level to benefit from the AWS failover handling. AWS takes care of promoting different database nodes to be the write node.              |
```
Comment (adityadav1987) [inline source/configure/database-configuration-settings.rst:146]: to be the "writer" node

File: source/scale/high-availability-cluster.rst [hunk 2]
```diff
@@ -216,9 +211,23 @@ Migrating to NAS or S3 from local storage is beyond the scope of this document.
 Database configuration
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Use the read replica feature to scale the database. The Mattermost server can be set up to use one master database and one or more read replica databases.
+.. tip::
+
+  Specifying configuration setting values using Mattermost environment variables ensure that they always take precedent over any other configuration settings.
+
+For an AWS High Availability RDS cluster deployment, point the `datasource </configure/environment-configuration-settings.html#database-datasource>`__ configuration setting to the write/read endpoint at the **cluster** level to benefit from the AWS failover handling. AWS takes care of promoting different database nodes to be the write node. Mattermost doesn't need to manage this.
```
Comment (adityadav1987) [inline source/scale/high-availability-cluster.rst:218]: "writer node"

File: source/scale/high-availability-cluster.rst [hunk 3]
```diff
@@ -216,9 +211,23 @@ Migrating to NAS or S3 from local storage is beyond the scope of this document.
...
+
+  Specifying configuration setting values using Mattermost environment variables ensure that they always take precedent over any other configuration settings.
+
+For an AWS High Availability RDS cluster deployment, point the `datasource </configure/environment-configuration-settings.html#database-datasource>`__ configuration setting to the write/read endpoint at the **cluster** level to benefit from the AWS failover handling. AWS takes care of promoting different database nodes to be the write node. Mattermost doesn't need to manage this. 
+
+Use the `read replica </configure/environment-configuration-settings.html#database-readreplicas>`__ feature to scale the database. The Mattermost server can be set up to use one master database and one or more read replica databases. 
+
+.. note::
+  
+  For an AWS High Availability RDS cluster deployment, don't hard-code the IP addresses. Point this configuration setting to the write/read endpoint at the **cluster** level. This will benefit from the AWS failover handling where AWS takes care of promoting different database nodes to be the write node. Mattermost doesn't need to manage this.
```
Comment (adityadav1987) [inline source/scale/high-availability-cluster.rst:224]: "writer" node

**Review by adityadav1987 — COMMENTED** [review]
Looks good to me. Just requested "writer" node to be mentioned instead of "write" node on multiple instances

**Comment by cwarnermm** [comment]
@adityadav1987 - Are you comfortable with this docs PR being merged and the content published to production?

**Pull request merged**